sdk/cosmos/azure_data_cosmos/src/options/mod.rs

```diff
@@ -111,3 +231,71 @@ pub struct ReadDatabaseOptions<'a> {
 pub struct ThroughputOptions<'a> {
     pub method_options: ClientMethodOptions<'a>,
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
what other test cases can we think of? like having few of them as None while others are filled in? or maybe having invalid values for some of them? like invalid types if possible?

Reply:
I don't think we'd want to validate things like session tokens or etags. Are there any other invalid values that could end up here given that we're using statically typed enums and lists?

Reply:
Our `as_headers` implementation is infallible, there aren't any errors it can produce right now. Which makes sense as it's largely just a "plain-old-data" type that doesn't validate it's properties. The `as_headers` just converts that to headers and lets the server sort it out if there's a problem.

The `AsHeaders` trait _requires_ that we return a `Result<>` and specify an error type, because it needs to be able to support implementations that _do_ return errors, but we use the `Infallible` type (which represents no possible errors) in our implementation